lib/Interpreter/IncrementalParser.cpp

```diff
@@ -182,22 +192,32 @@ namespace cling {
     initializeVirtualFile();
   }
 
-  void
+  bool
   IncrementalParser::Initialize(llvm::SmallVectorImpl<ParseResultTransaction>&
                                 result, bool isChildInterpreter) {
     m_TransactionPool.reset(new TransactionPool(getCI()->getSema()));
+    if (!m_TransactionPool) {
+      llvm::errs() << "TransactionPool could not be created.\n";
+      return false;
+    }
+
     if (hasCodeGenerator()) {
       getCodeGenerator()->Initialize(getCI()->getASTContext());
       m_BackendPasses.reset(new BackendPasses(getCI()->getCodeGenOpts(),
                                               getCI()->getTargetOpts(),
                                               getCI()->getLangOpts()));
+      if (!m_BackendPasses) {
```

Comment:
I generally agree with all of this. My only issue is that if Interpreter::Interpreter is the only user of IncrementalParser::Initialize I know of, and if Interpreter::Interpreter fails for whatever reason, I need users to be able to continue working until they decide to quit, no segfault, no aborting.

Unless libstd++ or libc++ was built with -f-no-exceptions, then any failure of new should be throwing std::bad_alloc anyway.

As the code was it was blindly assuming everything succeeded, so could wind up dereference null pointer which is a crash. And asserting won't change this behavior when built in Release.